.github/workflows/run-tests.yml

```diff
@@ -27,6 +28,52 @@ jobs:
       - name: Run Jest Tests
         run: yarn test --collectCoverage=false
 
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Branch Name
+        id: get_branch  # Add an ID to this step to reference its output
+        run: |
+          branch_name=${{ github.head_ref }}
+
+          # Sanitize the branch name
+          sanitized_branch_name=$(echo "$branch_name" | sed -e 's/\//_/g' | tr '[:upper:]' '[:lower:]')
+
+          echo "::set-output name=branch_name::$sanitized_branch_name"
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push tag
+        if: ${{ steps.get_branch.outputs.branch_name != 'main' }}
+        uses: docker/build-push-action@v5
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
+          file: docker/production/Dockerfile
+          tags: jembi/smart-health-links-portal:${{ steps.get_branch.outputs.branch_name }}
+
+      - name: Build and push latest
+        if: ${{ steps.get_branch.outputs.branch_name == 'main' }}
+        uses: docker/build-push-action@v5
+        with:
+          platforms: linux/amd64,linux/arm64
+          push: true
```

Comment:
Should we not check if it is nit on pull request before we push the image?

Reply:
@jacob-khoza-symb Yup thats correct, I still need to make some updates on this action script.
The current changes are just intended for testing purposes to have an image build and sent to dockerhub to verify the build works as intended (without having to merge into master multiple times)

Side note: The entire GHA needs a bit of refactoring and improvement